borg/key.py

```diff
@@ -126,28 +126,29 @@ class AESKeyBase(KeyBase):
     def id_hash(self, data):
         """Return HMAC hash using the "id" HMAC key
         """
-        return HMAC(self.id_key, data, sha256).digest()
+        return hmac_sha256(self.id_key, data)
 
     def encrypt(self, data):
         data = self.compressor.compress(data)
         self.enc_cipher.reset()
         data = b''.join((self.enc_cipher.iv[8:], self.enc_cipher.encrypt(data)))
```

Comment:
We could also avoid this copy, since the outgoing buffer is controlled by AES. I.e. we could make the buffer 8 bytes larger in the first place and pass out+8 to EVP_EncryptUpdate.